cpp/src/arrow/sparse_tensor.cc

```diff
@@ -503,13 +639,40 @@ Status MakeSparseTensorFromTensor(const Tensor& tensor,
     case SparseTensorFormat::CSC:
       return MakeSparseTensorFromTensor<SparseCSCIndex>(tensor, index_value_type, pool,
                                                         out_sparse_index, out_data);
+    case SparseTensorFormat::CSF:
+      return MakeSparseTensorFromTensor<SparseCSFIndex>(tensor, index_value_type, pool,
+                                                        out_sparse_index, out_data);
+
     // LCOV_EXCL_START: ignore program failure
     default:
       return Status::Invalid("Invalid sparse tensor format");
       // LCOV_EXCL_STOP
   }
 }
 
+template <typename TYPE, typename IndexValueType>
+void assign_values(int64_t dimension, int64_t offset, int64_t first_ptr, int64_t last_ptr,
```

Comment:
This function should be put in the anonymous namespace.

Reply:
And could you please rename it to more specific name.
For example, since this function expands the sparse tensor into the dense tensor, one of the candidates can be `ExpandSparseCSFTensorValues`.

Reply:
Done.